src/EnergyPlus/HVACUnitarySystem.cc

```diff
@@ -2698,18 +2771,43 @@ namespace HVACUnitarySystem {
                         UnitarySystem(UnitarySysNum).HeatVolumeFlowRate(DesignSpecMSHP(MSHPIndex).NumOfSpeedHeating);
                 }
-                // these coil types do not have an idle speed air flow rate
-                UnitarySystem(UnitarySysNum).IdleVolumeAirRate = 0.0;
-                UnitarySystem(UnitarySysNum).IdleMassFlowRate = 0.0;
-                UnitarySystem(UnitarySysNum).IdleSpeedRatio = 0.0;
             }
+            if (UnitarySystem(UnitarySysNum).CoolCoilExists) {
+                if (allocated(UnitarySystem(UnitarySysNum).CoolVolumeFlowRate) && MSHPIndex > 0) {
+                    UnitarySystem(UnitarySysNum).IdleVolumeAirRate =
+                        min(UnitarySystem(UnitarySysNum).IdleVolumeAirRate,
+                            UnitarySystem(UnitarySysNum).HeatVolumeFlowRate(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio);
+                    UnitarySystem(UnitarySysNum).IdleMassFlowRate =
+                        min(UnitarySystem(UnitarySysNum).IdleMassFlowRate,
+                            UnitarySystem(UnitarySysNum).HeatMassFlowRate(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio);
+                    UnitarySystem(UnitarySysNum).IdleSpeedRatio =
+                        min(UnitarySystem(UnitarySysNum).IdleSpeedRatio, UnitarySystem(UnitarySysNum).MSHeatingSpeedRatio(NumSpeeds));
+                } else {
+                    UnitarySystem(UnitarySysNum).IdleVolumeAirRate = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirVolFlow;
+                    UnitarySystem(UnitarySysNum).IdleMassFlowRate = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirMassFlow;
+                    UnitarySystem(UnitarySysNum).IdleSpeedRatio = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirVolFlow / UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds);
+                }
+            } else if (MSHPIndex > 0) {
+                UnitarySystem(UnitarySysNum).IdleVolumeAirRate =
+                    UnitarySystem(UnitarySysNum).HeatVolumeFlowRate(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
+                UnitarySystem(UnitarySysNum).IdleMassFlowRate =
+                    UnitarySystem(UnitarySysNum).HeatMassFlowRate(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
+                UnitarySystem(UnitarySysNum).IdleSpeedRatio =
+                    UnitarySystem(UnitarySysNum).MSHeatingSpeedRatio(NumSpeeds) * DesignSpecMSHP(MSHPIndex).NoLoadAirFlowRateRatio;
+            } else {
+                UnitarySystem(UnitarySysNum).IdleVolumeAirRate = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirVolFlow;
+                UnitarySystem(UnitarySysNum).IdleMassFlowRate = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirMassFlow;
+                UnitarySystem(UnitarySysNum).IdleSpeedRatio = UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirVolFlow / UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(NumSpeeds);
```

Comment:
This looks like it's a cut/paste error since we are in the heating coil section. Now I think what IdleSpeedRatio means. It's a scalar, not dependent on heating or cooling. So this should probably be nocoolheatflow to fan flow? to max flow? This is all very convoluted.

Reply:
In Init this is done:

    if (MyFanFlag(UnitarySysNum)) {
        if (UnitarySystem(UnitarySysNum).ActualFanVolFlowRate != AutoSize) {
            if (UnitarySystem(UnitarySysNum).ActualFanVolFlowRate > 0.0) {
                UnitarySystem(UnitarySysNum).HeatingFanSpeedRatio =
                        UnitarySystem(UnitarySysNum).MaxHeatAirVolFlow / UnitarySystem(UnitarySysNum).ActualFanVolFlowRate;
                UnitarySystem(UnitarySysNum).CoolingFanSpeedRatio =
                        UnitarySystem(UnitarySysNum).MaxCoolAirVolFlow / UnitarySystem(UnitarySysNum).ActualFanVolFlowRate;
                UnitarySystem(UnitarySysNum).NoHeatCoolSpeedRatio =
                        UnitarySystem(UnitarySysNum).MaxNoCoolHeatAirVolFlow / UnitarySystem(UnitarySysNum).ActualFanVolFlowRate;

So why not initialize these speed ratio variables here?